tests/decode.cpp

```diff
@@ -33,25 +30,35 @@
 
 SharedPtr<VppInput> createInput(DecodeParameter& para, SharedPtr<NativeDisplay>& display)
 {
-    SharedPtr<VppInput> input(VppInput::create(para.inputFile, para.renderFourcc, para.width, para.height));
+    SharedPtr<VppInput> input(VppInput::create(para.inputFile, para.renderFourcc, para.width, para.height, para.useCAPI));
     if (!input) {
         fprintf(stderr, "VppInput create failed.\n");
         return input;
     }
-#ifdef __ENABLE_CAPI__
-    SharedPtr<VppInputDecodeCapi> inputDecode = std::tr1::dynamic_pointer_cast<VppInputDecodeCapi>(input);
-#else
-    SharedPtr<VppInputDecode> inputDecode = std::tr1::dynamic_pointer_cast<VppInputDecode>(input);
-#endif
-    if (!inputDecode) {
-        input.reset();
-        fprintf(stderr, "createInput failed, cannot convert VppInput to VppInputDecode\n");
-        return input;
-    }
-    if (!inputDecode->config(*display)) {
-        input.reset();
-        fprintf(stderr, "VppInputDecode config failed.\n");
-        return input;
+    if(para.useCAPI){
+        SharedPtr<VppInputDecodeCapi> inputDecode = std::tr1::dynamic_pointer_cast<VppInputDecodeCapi>(input);
+        if (!inputDecode) {
+            input.reset();
+            fprintf(stderr, "createInput failed, cannot convert VppInput to VppInputDecode\n");
+            return input;
+        }
+        if (!inputDecode->config(*display)) {
+            input.reset();
+            fprintf(stderr, "VppInputDecode config failed.\n");
+            return input;
+        }
+    }else{
+        SharedPtr<VppInputDecode> inputDecode = std::tr1::dynamic_pointer_cast<VppInputDecode>(input);
```

Comment:
thanks @xuguangxin , but the member function config() is not a virtual function.